paper/src/main/java/com/denizenscript/denizen/paper/properties/PaperWorldExtensions.java

```diff
@@ -23,6 +29,81 @@ public static void register() {
             return new ElementTag(world.getWorld().getNoTickViewDistance());
         });
 
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
+
+            // <--[tag]
+            // @attribute <WorldTag.gateway_count>
+            // @returns ElementTag(Number)
+            // @description
+            // Returns the number of end gateway portals.
+            // Only works in end worlds.
+            // -->
+            WorldTag.tagProcessor.registerTag(ElementTag.class, "gateway_count", (attribute, object) -> {
+                DragonBattle battle = object.getWorld().getEnderDragonBattle();
+                if (battle == null) {
+                    attribute.echoError("Provided world is not an end world!");
+                    return null;
+                }
+                return new ElementTag(battle.getGatewayCount());
+            });
+
+            // <--[tag]
+            // @attribute <WorldTag.healing_crystals>
+            // @returns ListTag(EntityTag)
+            // @description
+            // Returns a ListTag of the healing crystals located on top of the obsidian towers.

```

Comment:
I'd mention ender/end crystals as a search aid or in the description somewhere, healing crystals is technically right but most people probably won't know them by that name.